### ISSUE TITLE
Results include people I already follow

### ISSUE THREAD

**Issue by ppamidimarri** [gabipurcaru/followgraph#17]
Thank you for a great tool! 

I don’t know if it’s just me, but my results include people I already follow. If this is by design, would you please consider adding an option to exclude accounts that I already follow?

**Comment by RiderOfGiraffes** [comment]
Seconded ... the first 20 accounts it recommended/suggested were accounts I already follow, which renders it largely useless.

Implementing the exclusion of those accounts already followed would make this *really* useful.

**Comment by gabipurcaru** [comment]
@ppamidimarri, @RiderOfGiraffes: can you send me the Mastodon handles you've tried using? That would help debug things.

Alternatively, how many people do you follow? Is it more than 2000?

**Comment by RiderOfGiraffes** [comment]
When you say it would help debug thing ... are you already trying to exclude people who are already followed?

I'm probably not typical, and the results have recently changed. I'm ColinTheMathmo@mathstodon.xyz

I've just tried it again and of the top 10 recommended, I already follow 4.

And I do follow a *lot* of people, mostly on my own instance.

Let me know if I can help.

**Comment by gabipurcaru** [comment]
@RiderOfGiraffes, yes, I see that you follow way more than 2k people, which is where the code cuts off. I don't think it's feasible to fix that since the Mastodon API requires fetching the list paginated by 50 items or so (so 20 requests per 1k follows, then doing at least one request for each of those). 

I could look into adding a warning though, to explain why some of the users in the list are already followed; if anyone reads this message and wants to take a stab at a PR, I can support merging it.

**Comment by RiderOfGiraffes** [comment]
I'm guessing this is mostly of use to people who are only just getting started, so it's not really aimed at me anyway.  So if it's already trying to exclude people who are already being followed, then the behaviour is reasonable.  As you say, it might be worth putting up a warning that for people already following lots of accounts, it's impractical to check recommendations against followed accounts.

I hope that makes sense ... and thank you for the tool.  I've bookmarked it to recommend to others.

**Comment by LenzGr** [comment]
Came here to report the same - the list returned mostly people I already followed. I, too follow quite a large number of accounts (~3.2k)

**Comment by ppamidimarri** [comment]
> @ppamidimarri, @RiderOfGiraffes: can you send me the Mastodon handles you've tried using? That would help debug things.
> 
> Alternatively, how many people do you follow? Is it more than 2000?

Yes, I follow over 2k accounts. I am charvaka@hivemind.plus, on a tiny instance that I run.

**Comment by roycewilliams** [comment]
Naively, if a high-follow requesting account is detected, could the user be warned _and_ be directed to how to download their own follow list and then upload it to followgraph, to enrich the analysis?
